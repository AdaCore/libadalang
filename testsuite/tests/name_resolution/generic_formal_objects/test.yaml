driver: name-resolution
input_sources: [genform.adb]
