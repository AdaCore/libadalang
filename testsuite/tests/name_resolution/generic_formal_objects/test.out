Analyzing genform.adb
#####################

Resolving xrefs for node <GenericPackageInstantiation ["Inst1"] genform.adb:19:4-19:38>
***************************************************************************************

Expr: <Id "GF1" genform.adb:19:25-19:28>
  references:    <DefiningName "GF1" genform.adb:10:12-10:15>
  type:          None
  expected type: None
Expr: <Id "A" genform.adb:19:30-19:31>
  references:    <DefiningName "A" genform.adb:9:7-9:8>
  type:          None
  expected type: None
Expr: <Id "I" genform.adb:19:35-19:36>
  references:    <DefiningName "I" genform.adb:6:4-6:5>
  type:          <ConcreteTypeDecl ["R"] genform.adb:2:4-4:15>
  expected type: <ConcreteTypeDecl ["R"] genform.adb:2:4-4:15>

Resolving xrefs for node <GenericPackageInstantiation ["Inst2"] genform.adb:22:4-22:54>
***************************************************************************************

Expr: <Id "GF2" genform.adb:22:25-22:28>
  references:    <DefiningName "GF2" genform.adb:16:12-16:15>
  type:          None
  expected type: None
Expr: <Id "T" genform.adb:22:30-22:31>
  references:    <DefiningName "T" genform.adb:14:12-14:13>
  type:          None
  expected type: None
Expr: <Id "R" genform.adb:22:35-22:36>
  references:    <DefiningName "R" genform.adb:2:9-2:10>
  type:          None
  expected type: None
Expr: <Id "A" genform.adb:22:38-22:39>
  references:    <DefiningName "A" genform.adb:15:7-15:8>
  type:          None
  expected type: None
Expr: <Aggregate genform.adb:22:43-22:52>
  type:          <ConcreteTypeDecl ["R"] genform.adb:2:4-4:15>
  expected type: <ConcreteTypeDecl ["R"] genform.adb:2:4-4:15>
Expr: <Id "B" genform.adb:22:44-22:45>
  references:    <DefiningName "B" genform.adb:3:7-3:8>
  type:          None
  expected type: None
Expr: <Int genform.adb:22:49-22:51>
  references:    None
  type:          <ConcreteTypeDecl ["Universal_Int_Type_"] __standard:116:3-116:45>
  expected type: <ConcreteTypeDecl ["Integer"] __standard:4:3-4:54>


Done.
